full-service/src/db/account.rs

```diff
@@ -694,6 +680,41 @@ impl AccountModel for Account {
 
         let next_subaddress_index =
             next_subaddress_index.unwrap_or(DEFAULT_NEXT_SUBADDRESS_INDEX) as i64;
+        let default_public_address =
+            default_public_address.unwrap_or_else(|| view_account_key.default_subaddress());
+
+        if default_public_address.view_public_key()
+            != &view_account_key.default_subaddress_view_public()
+            || default_public_address.spend_public_key()
+                != &view_account_key.default_subaddress_spend_public()
```

Comment:
I don't think it's that expensive, but this is computing the view account public address potentially 3 times
```suggestion
        let next_subaddress_index =
            next_subaddress_index.unwrap_or(DEFAULT_NEXT_SUBADDRESS_INDEX) as i64;
        let view_public_address = view_account_key.default_subaddress();
        let default_public_address =
            default_public_address.as_ref().unwrap_or(&view_public_address);

        if default_public_address.view_public_key()
            != view_public_address.view_public_key()
            || default_public_address.spend_public_key()
                != view_public_address.spend_public_key()
```